problem using os.ls and os.chdir

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 11.4, 20F71
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz
CC version: Apple clang version 12.0.5 (clang-1205.0.22.11)

getwd: /Users/9xn/sh311.v
vmodules: /Users/9xn/.vmodules
vroot: /Users/9xn/v
vexe: /Users/9xn/v/v
vexe mtime: 2022-05-21 05:56:42
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 f6a0c26.a7afb2d

Git version: git version 2.30.1 (Apple Git-130)
Git vroot status: weekly.2022.09-685-ga7afb2d1 (22 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 46662e20

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg sh311.v`
```v
module main

import utilities.table as t_print
import os

fn list_files() ?[][]string {
	mut unchecked_files := os.ls('utilities/modules')?
	println(unchecked_files)
	for files in unchecked_files {
		println(files)
		if os.is_file(files) == true {
			println(files)
			//continue
		} else {
			println(os.ls(os.chdir(files)?)?)
			// if os.is_file(files) == true {
			// println(files)
			// continue
		//} 
		}
		println(files)
	}
		mut modules := [['Module:', 'Path:', 'Description']]
	return modules
}

fn main() {
	mut data := [
		['Module:', 'Path:', 'Description']
	]
	mods := list_files() or { [['null', 'null', 'null'],] }
	for i, mod in mods {
	//println('$i) $mod')
	data << mods//[['$i) $mod']]
}
	t := t_print.Table{
		data: data
		style: .simple
		header_style: .bold
		align: .left
		orientation: .row
		padding: 1
		tabsize: 4
	}
	println(t)
}
```

**What did you expect to see?**

for println to print out the files in the directory if it was a directory


**What did you see instead?**
```
utilities/table/table.v:48:6: warning: unused variable: `i`
   46 |     mut rowstrings := []string{}
   47 |     sc := get_style_config(t.style)
   48 |     for i, row in rowdata {
      |         ^
   49 |         mut styled_row := row.clone()
   50 |         rspace := get_row_spaces(row, colmaxes)
sh311.v:32:6: warning: unused variable: `i`
   30 |     ]
   31 |     mods := list_files() or { [['null', 'null', 'null'],] }
   32 |     for i, mod in mods {
      |         ^
   33 |     //println('$i) $mod')
   34 |     data << mods//[['$i) $mod']]
sh311.v:32:9: warning: unused variable: `mod`
   30 |     ]
   31 |     mods := list_files() or { [['null', 'null', 'null'],] }
   32 |     for i, mod in mods {
      |            ~~~
   33 |     //println('$i) $mod')
   34 |     data << mods//[['$i) $mod']]
==================
   ^~~~~~~~~~~~~~~~
/tmp/v_501/sh311.3245210859263881815.tmp.c:16652:39: error: too few arguments to function call, single argument 'path' was not specified
                        _option_Array_string _t4 = os__ls();
                                                   ~~~~~~ ^
/tmp/v_501/sh311.3245210859263881815.tmp.c:15806:22: note: 'os__ls' declared here
_option_Array_string os__ls(string path) {
                     ^
6 warnings and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -897,15 +897,6 @@ pub fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) 
 		}
 		c.check_expected_call_arg(arg_typ, c.unwrap_generic(param.typ), node.language,
 			call_arg) or {
-			// str method, allow type with str method if fn arg is string
-			// Passing an int or a string array produces a c error here
-			// Deleting this condition results in propper V error messages
-			// if arg_typ_sym.kind == .string && typ_sym.has_method('str') {
-			// continue
-			// }
-			if arg_typ_sym.kind == .void && param_typ_sym.kind == .string {
-				continue
-			}
 			if param.typ.has_flag(.generic) {
 				continue
 			}
@@ -990,7 +981,6 @@ pub fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) 
 		if func.language != .c && !c.inside_unsafe && arg_typ.nr_muls() != param.typ.nr_muls()
 			&& !(call_arg.is_mut && param.is_mut) && !(!call_arg.is_mut && !param.is_mut)
 			&& param.typ !in [ast.byteptr_type, ast.charptr_type, ast.voidptr_type] {
-			// sym := c.table.sym(typ)
 			c.warn('automatic referencing/dereferencing is deprecated and will be removed soon (got: $arg_typ.nr_muls() references, expected: $param.typ.nr_muls() references)',
 				call_arg.pos)
 		}

--- a/vlib/v/checker/tests/fn_call_arg_mismatch_err_c.out
+++ b/vlib/v/checker/tests/fn_call_arg_mismatch_err_c.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/fn_call_arg_mismatch_err_c.vv:13:18: error: `os.chdir(files) ?` (no value) used as value in argument 1 to `os.ls`
+   11 |             println(files)
+   12 |         } else {
+   13 |             println(os.ls(os.chdir(files)?)?)
+      |                           ~~~~~~~~~~~~~~~~
+   14 |         }
+   15 |         println(files)

--- a/vlib/v/checker/tests/fn_call_arg_mismatch_err_c.vv
+++ b/vlib/v/checker/tests/fn_call_arg_mismatch_err_c.vv
@@ -1,0 +1,29 @@
+module main
+
+import os
+
+fn list_files() ?[][]string {
+	mut unchecked_files := os.ls('utilities/modules')?
+	println(unchecked_files)
+	for files in unchecked_files {
+		println(files)
+		if os.is_file(files) == true {
+			println(files)
+		} else {
+			println(os.ls(os.chdir(files)?)?)
+		}
+		println(files)
+	}
+	mut modules := [['Module:', 'Path:', 'Description']]
+	return modules
+}
+
+fn main() {
+	mut data := [
+		['Module:', 'Path:', 'Description'],
+	]
+	mods := list_files() or { [['null', 'null', 'null']] }
+	for _, mod in mods {
+		data << mod
+	}
+}

